src/fastify-plugins/maps/index.js

```diff
@@ -20,57 +24,153 @@ export const plugin = fp(mapsPlugin, {
 /**
  * @typedef {object} MapsPluginOpts
  * @property {string} [prefix]
+ * @property {string} [defaultOnlineStyleUrl]
  */
 
 /** @type {import('fastify').FastifyPluginAsync<MapsPluginOpts>} */
 async function mapsPlugin(fastify, opts) {
   fastify.register(routes, {
     prefix: opts.prefix,
+    defaultOnlineStyleUrl: opts.defaultOnlineStyleUrl,
   })
 }
 
+const GetStyleJsonQueryStringSchema = T.Object({
+  key: T.Optional(T.String()),
+})
+
 /** @type {import('fastify').FastifyPluginAsync<MapsPluginOpts, import('fastify').RawServerDefault, import('@fastify/type-provider-typebox').TypeBoxTypeProvider>} */
-async function routes(fastify) {
-  fastify.get('/style.json', async (req, rep) => {
-    const serverAddress = await getFastifyServerAddress(req.server.server)
-
-    // 1. Attempt to get "default" local static map's style.json
-    {
-      const styleId = 'default'
-
-      const results = await Promise.all([
-        fastify.mapeoStaticMaps.getStyleJsonStats(styleId),
-        fastify.mapeoStaticMaps.getResolvedStyleJson(styleId, serverAddress),
-      ]).catch(() => {
-        fastify.log.warn('Cannot read default static map')
-        return null
-      })
-
-      if (results) {
+async function routes(fastify, opts) {
+  const { defaultOnlineStyleUrl = DEFAULT_MAPBOX_STYLE_URL } = opts
+
+  fastify.get(
+    '/style.json',
+    { schema: { querystring: GetStyleJsonQueryStringSchema } },
+    async (req, rep) => {
+      const serverAddress = await getFastifyServerAddress(req.server.server)
+
+      // 1. Attempt to get "default" local static map's style.json
+      {
+        const styleId = 'default'
+
+        const results = await Promise.all([
+          fastify.mapeoStaticMaps.getStyleJsonStats(styleId),
+          fastify.mapeoStaticMaps.getResolvedStyleJson(styleId, serverAddress),
+        ]).catch(() => {
+          fastify.log.warn('Cannot read default static map')
+          return null
+        })
+
+        if (results) {
+          const [stats, styleJson] = results
+          rep.headers(createStyleJsonResponseHeaders(stats.mtime))
+          return styleJson
+        }
+      }
+
+      // 2. Attempt to get a default style.json from online source
+      {
+        const { key } = req.query
+        const upstreamUrl = key
+          ? styleUrlWithApiKey(defaultOnlineStyleUrl, key)
+          : defaultOnlineStyleUrl
+
+        try {
+          const upstreamResponse = await fetch(upstreamUrl)
+
+          if (upstreamResponse.ok) {
+            // Set up headers to forward
+            for (const [name, value] of upstreamResponse.headers) {
+              // TODO: Typically Mapbox gzips the content but we'd need to make sure
+              // something like https://github.com/fastify/fastify-compress/ is registered
```

Comment:
I don't see this as a TODO. I think we want to use our own compression regardless of what the upstream source says.

```suggestion
              // We do our own content encoding.
```

Reply:
I still consider it a TODO because there's no content-encoding implemented on our end yet. But yeah mostly added. the note to remind us that we should probably have something implemented

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/459/commits/af0d737ace5f9a0895fc696a26b77d469d19001a